client/plasma_cash_client.go

```diff
@@ -119,16 +119,15 @@ func (c *PlasmaCashClient) Deposit(deposit *pctypes.DepositRequest) error {
 
 // Sends a plasma cash transaciton to be added to the current plasma cash block
 func (c *PlasmaCashClient) SendTransaction(slot uint64, prevBlock *big.Int, denomination *big.Int,
-	newOwner, prevOwner string, sig []byte) error {
+	newOwner, prevOwner string, sig []byte, hash []byte) error {
 	receiverAddr := loom.MustParseAddress(fmt.Sprintf("eth:%s", newOwner))
-	senderAddr := loom.MustParseAddress(fmt.Sprintf("eth:%s", prevOwner))
 	tx := &pctypes.PlasmaTx{
 		Slot:          uint64(slot),
 		PreviousBlock: &types.BigUInt{Value: *loom.NewBigUInt(prevBlock)},
 		Denomination:  &types.BigUInt{Value: *loom.NewBigUInt(denomination)},
 		NewOwner:      receiverAddr.MarshalPB(),
-		Sender:        senderAddr.MarshalPB(),
 		Signature:     sig,
+		Hash:          hash,
```

Comment:
Hash slot + sender + nonce, sign, put sig as new field in `PlasmaTxRequest`

Reply:
@enlight In this case, we will also need to store that hash in a new field. As without the key, only thing we can validate with `signature` is `sender`, and we wont be able to verify if attacker has changed fields or not.

Reply:
@enlight Cool. So, this will be from the client side (As you are using private key there)

On contract side, I will be doing something along this lines:

1. Recover public key and address from signature.
2. Verify input hash by recalculating and comparing with input hash. 
2. Use public key (recovered), signature and hash (From plasmaTx) to verify if signature is valid and generated from same public key (Function is available to do that in crypto)
3. If all of this is successful set `sender` variable using sender address, otherwise, deny access.

Reply:
You don't need to deal with public keys:
```golang
senderAddr, err := evmcompat.RecoverAddressFromTypedSig(hash, sig)
if err != nil {
	return err
}
```

In the contract you'd:
1. Reconstruct the hash of the `PlasmaTx` (using `LoomTx` as you did initially), then use the hash & sig to recover the sender. Check recovered sender against `PlasmaTx.Sender`, you've mentioned you removed this field because it's redundant, but this check helps track down client-side signing bugs.
2. Reconstruct the second hash (sender, nonce, tx hash), and use `PlasmaTxRequest.Signature` to recover the sender again. Check nonce matches, check the recovered sender address matches `PlasmaTx.Sender`.